client/layout/index.jsx

```diff
@@ -100,6 +101,20 @@ class Layout extends Component {
 		// intentionally don't remove these in unmount
 	}
 
+	renderMasterBar() {
+		if ( 'jetpack-cloud' === config( 'masterbar' ) ) {
```

Comment:
I'd say one thing for the new `masterbar` config: YAGNI. Right now we always render the same masterbar for all Jetpack Cloud environments, so why not just check the environment here? And if the need comes for some more complex logic, we could always introduce it.

Reply:
I removed the `masterbar` config and it is just looking for the `jetpack-cloud` feature flag now.